openquake/hazardlib/shakemap.py

```diff
@@ -198,25 +197,24 @@ def cross_correlation_matrix(imts, corr='yes'):
     assert corr in 'yes no full', corr
     # if there is only PGA this is a 1x1 identity matrix
     M = len(imts)
-    cross_matrix = numpy.zeros((M, M))
-    for i, im in enumerate(imts):
-        T1 = im.period or 0.05
-
-        for j in range(M):
-            T2 = imts[j].period or 0.05
-            if i == j:
-                cross_matrix[i, j] = 1
-            else:
+    cross_matrix = numpy.eye(M)
+    if corr == 'full':
+        cross_matrix = numpy.full((M, M), 0.99999)
```

Comment:
We would need a comment here explaining the magic number  0.99999

Reply:
Took the magic number from line 214 of the original code. 

I can only guess that there is some numerical reason?

Reply:
You are right, it was there from the beginning. Let's me accept the PR as it is, then.